CorsixTH/Lua/dialogs/fullscreen/annual_report.lua

```diff
@@ -133,39 +133,28 @@ function UIAnnualReport:UIAnnualReport(ui, world)
 
   -- Get and sort values used on the statistics screen.
   -- The six categories. The extra tables are used to be able to sort the values.
-    self.money = {}
     self.money_sort = {}
-    self.visitors = {}
     self.visitors_sort = {}
-    self.salary = {}
     self.salary_sort = {}
-    self.deaths = {}
     self.deaths_sort = {}
-    self.cures = {}
     self.cures_sort = {}
-    self.value = {}
     self.value_sort = {}
 
     -- TODO: Right now there are no real competitors, they all have initial values.
     for i, hospital in ipairs(world.hospitals) do
-      self.money[hospital.name] = hospital.balance - hospital.loan
-      self.money_sort[i] = hospital.balance - hospital.loan
-      self.visitors[hospital.name] = hospital.num_visitors
-      self.visitors_sort[i] = hospital.num_visitors
-      self.deaths[hospital.name] = hospital.num_deaths
-      self.deaths_sort[i] = hospital.num_deaths
-      self.cures[hospital.name] = hospital.num_cured
-      self.cures_sort[i] = hospital.num_cured
-      self.value[hospital.name] = hospital.value
-      self.value_sort[i] = hospital.value
-      self.salary[hospital.name] = hospital.player_salary
-      self.salary_sort[i] = hospital.player_salary
+      self.money_sort[i] = {hospital.balance - hospital.loan, i}
+      self.visitors_sort[i] = {hospital.num_visitors, i}
+      self.deaths_sort[i] = {hospital.num_deaths, i}
+      self.cures_sort[i] = {hospital.num_cured, i}
+      self.value_sort[i] = {hospital.value, i}
+      self.salary_sort[i] = {hospital.player_salary, i}
     end
 
-    local sort_order = function(a,b) return a>b end
+    local sort_order = function(a,b) return a[1] > b[1] or (a[1] == b[1] and a[2] == 1) end
```

Comment:
Naming nit: asc_order / desc_order. Also comment that it keeps the player on top for equivalence, that was not obvious.
Naming the keys and giving the player_hospital a variable would reduce the number of magic numbers and make the code more readable IMO:

```lua
for i, hospital in ipairs(world.hospitals) do
  self.money_sort[i] = { value: hospital.balance - hospital.loan, hosp_index: i }
  ...
end

-- When sorting keep the player hospital on top when values are the same
local player_index = 1
local desc_order = function(a, b) return a['value'] > b['value'] or (a['value' == b['value'] and a[hosp_index] == player_index) end
local asc_order = function(a, b) return a['value'] < b['value'] or (a['value'] == b['value'] and a[hosp_index] == player_index) end
```